module/VuFind/src/VuFind/UrlShortener/Database.php

```diff
@@ -133,25 +92,33 @@ protected function base62Decode($base62Number)
     public function shorten($url)
     {
         $path = str_replace($this->baseUrl, '', $url);
-        $this->table->insert(['path' => $path]);
-        $id = $this->table->getLastInsertValue();
+        $hash = hash(static::HASH_ALGO, $path . $this->salt);
+        $shorthash = substr($hash, 0, 9);
+        $results = $this->table->select(['hash' => $shorthash]);
+
+        // this should almost never happen - we then return the existing hash
+        if ($results->count() == 0) {
+            $this->table->insert(['path' => $path, 'hash' => $shorthash]);
+        }
 
-        $shortUrl = $this->baseUrl . '/short/' . $this->base62Encode($id);
-        return $shortUrl;
+        return $this->baseUrl . '/short/' . $shorthash;
     }
 
     /**
      * Resolve URL from Database via id.
      *
-     * @param string $id ID to resolve
+     * @param string $input hash
      *
      * @return string
+     *
+     * @throws Exception
      */
-    public function resolve($id)
+    public function resolve($input)
     {
-        $results = $this->table->select(['id' => $this->base62Decode($id)]);
+        $shorthash = substr($input, 0, 9);
```

Comment:
Is there any reason to substring the input here? Can't we just assume that whatever we output from shorten will get sent back to resolve? I think it gives us more flexibility if we make no specific assumptions about hash length in this step.

Reply:
There is no reason except saving disk space.